src/pebble.py

```diff
@@ -299,9 +315,12 @@ def _environment_has_changed(
 
 
 # The complexity of this method will be reviewed.
-def reconcile(  # noqa: C901 pylint: disable=too-many-branches,too-many-statements
+# pylint: disable=too-many-branches,too-many-statements
+# pylint: disable=too-many-arguments,too-many-positional-arguments
+def reconcile(  # noqa: C901
```

Comment:
I'm afraid of how big this reconcile is getting :-P

Reply:
We discussed with Thanh about this, this method definitely requires some refactoring